vmdb/app/models/ems_refresh/parsers/openstack_infra.rb

```diff
@@ -52,11 +53,22 @@ def hosts_ports
         @hosts_ports ||= @baremetal_service.ports.details
       end
 
+      def stacks
+        @stacks ||= detailed_stacks
+      end
+
+      def detailed_stacks
+        detailed_stacks = []
+        # Seems like Heat doesn't suport detailed list, so we need to call get for each stack, which is suboptimal
+        @orchestration_service.stacks.each { |stack| detailed_stacks << stack.details }
+        detailed_stacks
```

Comment:
Looks like a good opportunity to use `each_with_object`:

```ruby
  def details_stacks
    @orchestration_service.stacks.each_with_object([]) { |stack, detailed_stacks| detailed_stacks << stack.details }
  end
```

Reference: http://rubyquicktips.com/post/20104758978/enumerable-each-with-object